### DISCUSSION TITLE
How can I make the keepassxc clear the password from the cliphist?

### DISCUSSION THREAD

**Discussion by azizsayfiddin** [keepassxreboot/keepassxc#10704] [Q&A] [answered]
The keepassxc doesn't delete the password from the [cliphist](https://github.com/sentriz/cliphist) after 10 seconds. I think it should run the `cliphist delete <password>` after 10 seconds to delete it from the cliphist.

**Comment by droidmonkey** [comment]
You need to tell cliphist to support the standard password mime data flags for clipboard.

**Reply by azizsayfiddin** [reply]
> You need to tell cliphist to support the standard password mime data flags for clipboard.

How can I do that?

**Reply by droidmonkey** [reply]
With a bug report to them

**Reply by winkelnp** [reply]
Does keepassxc use the `CLIPBOARD_STATE` protocol? when I watch its state when copying a password it seems to set `CLIPBOARD_STATE=data` in place of `CLIPBOARD_STATE=sensitive` or is this on some other end?

**Reply by droidmonkey** [reply]
Never seen that before. Link to definition of that "protocol"?

**Reply by winkelnp** [reply]
It's an (in their words) "mini-protocol" that wl-clipboard gives, explained in their man page: https://man.archlinux.org/man/wl-clipboard.1

As such I'm not sure how it would have to be implemented to work or if that would even be on something like a password manager to support, and it is, of course, everything but official for wayland, just hopeful for a future wayland-protocol


Complete aside: by "standard mime data flags" do you mean `x-kde-passwordManagerHint`?

**Reply by droidmonkey** [reply]
> However, the currently existing Wayland clipboard protocols don't let wl-clipboard identify the cases where clear and sensitive values should be set, so currently wl-clipboard only ever sets CLIPBOARD_STATE to data or nil.

The KDE mime flag is much more of a standard than this environment variable.

**Reply by cnt0** [reply, marked as answer]
Hello, I did my research about this topic, and the current state is the following:
- `wl-clipboard` learned to check the `x-kde-passwordManagerHint` mime flag, and in this case it sets the `CLIPBOARD_STATE` environment variable to `sensitive` according to its mini-protocol
- `cliphist` supports this mini-protocol out of box
- `wl-clipboard` maintainer has yet to release a new version with this feature, so it's not in downstream repositories yet. So right now it's better to build it manually from the git master. [relevant issue](https://github.com/bugaevc/wl-clipboard/issues/260)

So I believe nothing technical needs to be done on the keepassxc side. Maybe it worth mentioning somewhere in the documentation that it's advised to use `wl-clipboard` with version >2.2.1 or from git master.
